plugins/inventory/gcp_compute.py

```diff
@@ -236,7 +236,7 @@ def hostname(self):
         """
         for order in self.hostname_ordering:
             name = None
-            if order.startswith("labels."):
+            if order.startswith("labels.") and "labels" in self.json:
```

Comment:
I think this needs to be:

```
if order.startswith("labels."):
    if "labels" in self.json:
        name = self.json["labels"].get(order[7:])
```

Otherwise when `"labels"` is not in `self.json` we'll hit the `raise` at line 248 rather than continue looping through `self.hostname_ordering` (since none of the `elif` cases will match).

Reply:
@SirGitsalot good catch, updated!